### PR TITLE
Box storage changes

### PR DESCRIPTION
**Pull request by jasonpaulos**
Child PR to #345.

This addresses most of my comments.

Still needs to be addresses:
- [x] https://github.com/algorand/java-algorand-sdk/pull/345#discussion_r976911364
  - Addressed by #401 
- [x] https://github.com/algorand/java-algorand-sdk/pull/345#discussion_r976865846
  - Addressed by #399
- [x] https://github.com/algorand/java-algorand-sdk/pull/345#discussion_r976866089
  - Addressed by #399

### CODE REVIEW

File: src/test/java/com/algorand/algosdk/integration/Applications.java [hunk 1]
```diff
@@ -275,12 +276,26 @@ else if (fromClient.equals("indexer"))
...
-    private static boolean contains(byte[] elem, List<byte[]> xs) {
-        for (byte[] e : xs) {
-            if (Arrays.equals(e, elem))
-                return true;
+    private static class ByteArrayComparator implements Comparator<byte[]> {
+        public ByteArrayComparator(){}
+
+        @Override
+        public int compare(byte[] a, byte[] b) {
+            return Arrays.compare(a, b);
```
Comment (jasonpaulos) [inline src/test/java/com/algorand/algosdk/integration/Applications.java:284]: This method is introduced in Java 9, so we'll need a workaround

File: src/main/java/com/algorand/algosdk/builder/transaction/ApplicationBaseTransactionBuilder.java [hunk 2]
```diff
@@ -103,10 +103,6 @@ private List<Transaction.BoxReference> convertBoxes(List<AppBoxReference> abrs,
         return xs;
     }
 
-    /**
-     * BoxReferences lists the boxes whose state may be accessed during the execution
-     * of this application call. The access is read-only.
-     */
```
Comment (jasonpaulos) [inline src/main/java/com/algorand/algosdk/builder/transaction/ApplicationBaseTransactionBuilder.java:109]: By deleting this comment, we'll inherit the comment from `ApplicationCallReferencesSetter`

File: src/test/java/com/algorand/algosdk/util/ComparableBytes.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+public class ComparableBytes {
+    private final byte[] data;
+
+    public ComparableBytes(byte[] data) {
+        this.data = data;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```
[the remark this answers is not in the document]
Comment (jasonpaulos) [reply src/test/java/com/algorand/algosdk/util/ComparableBytes.java]: Thanks for catching this, resolved in e84187959cb4950ec8aacad9b58fcd935a03e5b7

**Review by michaeldiamant — APPROVED** [review]

**Review by ahangsu — APPROVED** [review]
Thanks for the `ComparableBytes`, and the changes are looking good to me. There's a merge conflict, we are taking the `feature` branch's change, right?

**Comment by jasonpaulos** [comment]
@ahangsu thanks for pointing out the merge conflict, it's fixed now

**Review by michaeldiamant — APPROVED** [review]
@jasonpaulos Looks accurate post-merge conflict resolution.

**Pull request merged**